File: BubbleSort.cpp
#include <bits/stdc++.h>
#include <stdlib.h> 
#include<time.h>
#include <stdio.h> 
#include <chrono> 

using namespace std; 
using namespace std::chrono;
  
void swap(int *xp, int *yp)  
{  
    int temp = *xp;  
    *xp = *yp;  
    *yp = temp;  
}  
  
void bubbleSort(int *arr, int n)  
{  
    int i, j;  
    for (i = 0; i < n-1; i++)      
      
    for (j = 0; j < n-i-1; j++)  
        if (arr[j] > arr[j+1])  
            swap(&arr[j], &arr[j+1]);  
}  
  
void printArray(int *arr, int size)  
{  
    int i;  
    for (i = 0; i < size; i++)  
        cout << arr[i] << " ";  
    cout << endl;  
}

int main()  
{  
    int numOfInt = 0;
    cout << "Please enter the number of integers you would like to be produced at random then sorted: ";
    cin >> numOfInt;
    int arr [numOfInt];  
    
    srand(time(0));
    
    for (int i = 0; i < numOfInt; i++)
        arr[i] = rand();
    
    auto start = high_resolution_clock::now(); 
    bubbleSort(arr, numOfInt);  
    auto stop = high_resolution_clock::now(); 
    auto duration = duration_cast<nanoseconds>(stop - start);
    
    cout<<"Sorted array: \n";  
    printArray(arr, numOfInt);  
    
    cout << "Time taken: " << duration.count() << " ns" << endl;
    return 0;  
}

File: BubbleSort.java
import java.util.*;


public class BubbleSort{
    public static void main(String args[]){
        Scanner keyboard = new Scanner(System.in);
        System.out.println("Please enter the number of integers you would like to be produced at random then sorted");
        Random rand = new Random();
        int input = keyboard.nextInt();
        int arr [] = new int [input];

        for (int i = 0; i < input;i++)
            arr[i] = rand.nextInt(input);
        printArray(arr);
        long startTime = System.nanoTime();

        bubbleSort(arr);
        
        long endTime = System.nanoTime();

        printArray(arr);
        System.out.println("The programs runtime was: "+(endTime - startTime) + " ns."); 

        keyboard.close();

    }
    public static void bubbleSort(int arr[]){
        int temp = 0;
        for (int i = 0;i < arr.length-1;i++)
            {
                for (int j = 0; j < arr.length - 1;j++)
                {
                    if (arr[j] > arr[j + 1])
                    {
                        temp = arr[j+1];
                        arr[j+1] = arr[j];
                        arr[j] = temp;
                    }
                }
            }
        }
    
    public static void printArray(int arr[]){
        System.out.println("The array holds:");
        for (int i = 0;i < arr.length;i++)
            System.out.print(arr[i]+" ");
        System.out.println("");    
    }
}